### PR TITLE
Runner Reconnect: Connection Process Revamp

### PR DESCRIPTION
**Pull request by piotrek6641**
**What?**  

Change starting Instance flow to 

image

**Why?**

STH can be restarted without stopping Instances. Instances will reconnect when STH becomes available.

🧊 **Important**

This PR covers JS Runner only. Python runner can be started in new way but it can't reconnect after it has been disconnected. 

**Clickup Task:** 
https://app.clickup.com/t/24308805/VDM-385

**Review checks:**

These aspects need to be checked by the reviewer:

- [x] Verify and confirm operation (please post a screenshot) 
- [x] All STH tests pass
- [x] All [Scramjet Cloud Platform](https://docs.scramjet.org/platform) tests pass
- [x] Documentation is updated or no changes

### CODE REVIEW

**Comment by patuwwy** [comment]
requires changes in Scramjet framework (python) https://github.com/scramjetorg/framework-python/pull/44

**Comment by alicja-gruzdz** [comment]
Will be released together with
https://github.com/scramjetorg/transform-hub/pull/985
https://github.com/scramjetorg/transform-hub/pull/1002
as they include breaking changes.

**Review by kociolekscramjet — APPROVED** [review]
That's a lot of changes in the core functionalities. But it looks senile :D
No screenshot - everything should work as before but better :)

**Comment by a-tylenda** [comment]
Bugs fixed within this PR:
- STH Reconnect docker issue](https://app.clickup.com/t/24308805/VDM-1667)
- [STH Reconnect panel ("Instances" tab throws an error)](https://app.clickup.com/t/24308805/VDM-1669)

File: packages/adapters/src/kubernetes-instance-adapter.ts [hunk 1]
```diff
@@ -90,14 +95,13 @@ IComponent {
...
-
-    async run(config: InstanceConfig, instancesServerPort: number, instanceId: string): Promise<ExitCode> {
+    async dispatch(config: InstanceConfig, instancesServerPort: number, instanceId: string, sequenceInfo: SequenceInfo, payload: RunnerConnectInfo): Promise<void> {
         if (config.type !== "kubernetes") {
             throw new Error(`Invalid config type for kubernetes adapter: ${config.type}`);
         }
 
         if (this.adapterConfig.quotaName && await this.kubeClient.isPodsLimitReached(this.adapterConfig.quotaName)) {
-            return RunnerExitCode.PODS_LIMIT_REACHED;
+            throw Error(RunnerExitCode.PODS_LIMIT_REACHED.toString());
```
Comment (ErykSolecki) [inline packages/adapters/src/kubernetes-instance-adapter.ts:104]: This line probably breaks responses integrity returned from backend.

Check this issue to test response on frontend:
https://github.com/scramjetorg/cloud-platform-panel/issues/524

Old response from server was:
`{"error":"Instance limit reached"}`
Response after changes:
`{"error":"24"}`

**Review by ErykSolecki — CHANGES_REQUESTED** [review]

**Review by a-tylenda — APPROVED** [review]

**Pull request merged**